src/storage/src/hummock/compactor/mod.rs

```diff
@@ -225,6 +227,77 @@ impl Compactor {
             .await;
         let multi_filter_key_extractor = Arc::new(multi_filter_key_extractor);
 
+        let sstable_ids = compact_task
+            .input_ssts
+            .iter()
+            .flat_map(|level| level.table_infos.iter())
+            .map(|table_info| table_info.id)
+            .collect_vec();
+
+        let compaction_size = compact_task
+            .input_ssts
+            .iter()
+            .flat_map(|level| level.table_infos.iter())
+            .map(|table_info| table_info.file_size)
+            .sum::<u64>();
+
+        let mut indexes = vec![];
+
+        // preload the meta and get the smallest key to split sub_compaction
+        for sstable_id in sstable_ids {
```

Comment:
Please check `compaction_size` before and does not call `sstable` to avoid extra RPC